### ISSUE TITLE
Package is not available on pypi

### ISSUE THREAD

**Issue by olliebun** [nlhepler/pydot#12]
I want to publish a package on pypi that depends on `pydot`.

As you're aware, the [pypi package](https://pypi.python.org/pypi/pydot) is unmaintained an out-of-date.

I briefly had my own fork and contacted the maintainer. He seemed willing to pass on ownership of the palckage, but the conversation never went beyond that.

IMO uploading the package under a different name (even `pydot2`) would be a good way forward.

**Comment by johnyf** [comment]
I think that such PyPI entries already exist:
- [`pydot-ng`](https://pypi.python.org/pypi/pydot-ng/1.0.0)
- [`pydotplus`](https://pypi.python.org/pypi/pydotplus/2.0.2).

I believe that preserving the name is important, in order to avoid unnecessary renames in code, over many packages.
The original googlecode page of `pydot` has been updated to forward to the [original author's repository on github](https://github.com/erocarrera/pydot). So, an "official" repository exists on github. The next thing would be to:
1. merge the changes collected here to upstream.
2. find a maintainer for the PyPI entry.

I would be happy to do both, as I have had the same issue with `pydot` being an exception for `pip`. You mention that the PyPI owner appears willing to add a maintainer. So, there was just no further action in that direction ?

**Comment by olliebun** [comment]
Yep, I agree with all of that.

I was in contact with Ero Carrera over email about adding a maintainer, he seemed happy with it, I followed up to ask about the next steps and he didn't get back to me. Definitely worth trying again. I can reach out again if you want - or leave it to you. This was about 18 months ago IIRC.

**Comment by peterjc** [comment]
+1 for someone taking over the existing PyPI entry and merging these Python 3 fixes.

**Comment by takluyver** [comment]
Yes, it would definitely be good to try to consolidate the different pydot forks and update the package on PyPI.

Pinging @ksambor and @prmtl of pydot-ng, @carlos-jenkins of pydot-plus, and @ajsteven130, whose fork has recent work.

**Comment by ajsteven130** [comment]
Hello,

For my fork, I just went through and cleared all the errors to get pydot to install on Python 3.4. It installs and works fine, but I am not sure it should be counted on as an official package. I'd be very happy if the small of work there is useful and could be integrated into a better maintained repository.

If there is any way I can help, let me know

**Comment by johnyf** [comment]
[bmcorser](https://github.com/bmcorser) has a PyPI entry for Python 3 called [`pydot3k`](https://pypi.python.org/pypi/pydot3k/1.0.17). Note that it installs as `pydot`, so changes needed in source code. This enables [conditional dependency selection](https://github.com/johnyf/dd/blob/c09e71f675afdaa325595a6d1c10cb72fe5034a6/setup.py#L28), until the `pydot` entry is updated.

**Comment by prmtl** [comment]
Our fork (`pydot-ng`) was created because we needed the work from https://github.com/nlhepler/pydot and a package on PyPI (with support for Python 2.6). There are no much changes from original fork, mostly cleaning code/

**Comment by takluyver** [comment]
Just a reminder, it would be great if someone could work out which of the various forks is the best starting point, contact Ero to get the name on PyPI and upload a new release. I just helped [another person](http://stackoverflow.com/q/35828821/434217) on StackOverflow who was trying to use the original 'pydot' package on Python 3.

**Comment by johnyf** [comment]
I emailed @erocarrera (last week) about maintenance of `pydot` and its PyPI entry. I received no reply, thus I will evaluate the other packages (non-`pydot`) and switch to a maintained one.

The collection of `pydot` forks have various names and fixes, but are isolated side-projects that tend to become unmaintained projects too. For example, `pydotplus` is restructured (from a module to a package with a non-standard directory tree), and [doesn't define any `pydotplus.__version__`](https://github.com/carlos-jenkins/pydotplus/blob/e06552ea3989bfb998ef6920a07d6c966ff9b01b/lib/pydotplus/__init__.py) (`__version__` was originally defined inside the standalone module `pydot.py`), but in [`pydotplus.version`](https://github.com/carlos-jenkins/pydotplus/blob/e06552ea3989bfb998ef6920a07d6c966ff9b01b/lib/pydotplus/version.py#L33).

PEP 8 is not followed, and top-down parsing (`pyparsing`) doesn't scale to what bottom-up parsing can do (lex-yacc, i.e., `ply` and other parser generators). This seems to suggest using a newer and maintained package, for example [`graphviz`](https://github.com/xflr6/graphviz) (even if it doesn't have a parser, after all serves mostly as a backend).

For posterity, https://github.com/carlos-jenkins/pydotplus/issues/1.

**Comment by nlhepler** [comment]
Hey all,

I think moving to a maintained version of pydot is definitely the way to go. I modified this package to assist me in my graduate studies, but I'm afraid I don't have the bandwidth to maintain it. Either pydot-ng or pydotplus seems to be the way to go.

TL;DR: I don't have the bandwidth to maintain this package, please look at forks, including [pydot-ng](https://github.com/pydot/pydot-ng) and [pydotplus](https://github.com/carlos-jenkins/pydotplus).

**Comment by johnyf** [comment]
@erocarrera has added me as a maintainer for `pydot` (on GitHub and PyPI), so we can start updating the `master` branch, and make a release. I think that this addresses the OP.

**Comment by takluyver** [comment]
Excellent, thanks @johnyf

**Comment by johnyf** [comment]
`pydot` is being maintained, and the latest version available on PyPI. I would suggest closing this issue.
